tests/conftest.py

```diff
@@ -158,6 +158,7 @@ def __init__(self):
         print 'Gas Used: %s' % (self.chain.head_state.gas_used - startingGas)
         self.scalarMarket = self.createReasonableScalarMarket(self.universe, 40, self.cash)
         self.constants = self.uploadAndAddToController("solidity_test_helpers/Constants.sol")
+        self.chain.mine(1)
```

Comment:
`createSnapshot()` should be the correct place for the `chain.mine(1)` call.  Can you go into detail on why you believe this move is necessary?

Anytime someone wants to create a snapshot, they _must_ mine a block first.  By putting the call to mine inside the `createSnapshot` function we ensure that a block will always be mined before creating a snapshot.  By putting it here, snapshots created in other code paths will have to mine a block themselves.

Reply:
Also, extra calls to `mine` shouldn't cause any problems.

Reply:
Issue comes up when multiple tests use createSnapshot() the second test suit gets errors: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:205: in createSnapshot
    self.chain.mine(1)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ethereum.tools.tester.Chain object at 0x10e7c9290>, number_of_blocks = 1
coinbase = '\x82\xa9x\xb3\xf5\x96*[\tW\xd9\xee\x9e\xefG.\xe5[B\xf1'

    def mine(self, number_of_blocks=1, coinbase=a0):
        self.cs.finalize(self.head_state, self.block)
        set_execution_results(self.head_state, self.block)
        self.block = Miner(self.block).mine(rounds=100, start_nonce=0)
>       assert self.chain.add_block(self.block)
E       AssertionError

/usr/local/lib/python2.7/site-packages/ethereum/tools/tester.py:167: AssertionError

What I don't know is why.

Reply:
Hmm, I'll checkout your branch and take a look to see if I can help figure out what the root cause is.  That is odd.